TypeCobol/Compiler/Nodes/ParametersProfileNode.cs

```diff
@@ -40,43 +40,6 @@ public IList<ParameterDescription> Parameters
         /// <summary>TCRFUN_NO_RETURNING_FOR_PROCEDURES</summary>
         public bool IsProcedure { get { return ReturningParameter == null; } }
 
-
-
-        public override bool Equals(object other)
-        {
-            if (other == null || GetType() != other.GetType()) return false;
-            var o = other as ParametersProfileNode;
-            if (o == null) return false;
-            // instead of doing foreach(var mode in Tools.Reflection.GetValues<Passing.Mode>()) ...,
-            // only iterate over input+output+inout parameters: returning parameter does not have
-            // any impact in conflict between function profiles resolution
-            bool okay = true;
-            okay = AreEqual(InputParameters, o.InputParameters);
-            if (!okay) return false;
-            okay = AreEqual(InoutParameters, o.InoutParameters);
-            if (!okay) return false;
-            okay = AreEqual(OutputParameters, o.OutputParameters);
-            return okay;
-        }
-
-        private static bool AreEqual(IList<ParameterDescription> mine, IList<ParameterDescription> hers)
-        {
-            if (mine.Count != hers.Count) return false;
-            for (int c = 0; c < mine.Count; c++)
-            {
-                if (!mine[c].Equals(hers[c])) return false;
-                if (!mine[c].Equals(hers[c])) return false;
-            }
-            return true;
-        }
-
-        public override int GetHashCode()
-        {
-            int hash = 17;
-            foreach (var p in Parameters) hash = hash * 23 + p.GetHashCode();
-            return hash;
-        }
```

Comment:
Why delete the methods ?? The idea here is to have the IEquatable/Equals/GetHashCode and forward the implementation to the CodeElement because we already have an equality comparison mechanism for the ParametersProfile class.

Reply:
It's right; it's fixed.
https://github.com/TypeCobolTeam/TypeCobol/pull/1600/commits/8b7686fa9f02c00e2a7150589c65d7fd66b787c8